SecureAuth.Sdk/AuthenticationService.cs

```diff
@@ -349,10 +348,58 @@ public SendOtpResponse SendAdHocEmailOtp(AdHocEmailOtpRequest request)
                 throw new ArgumentNullException("AdHocEmailOtpRequest.Token", "Token cannot be empty");
             }
             // process request
-            return SendOtp(request);
+            return SendOtp(request, errorOnAccountStatus ? ApiVersion.V1 : ApiVersion.V2);
+        }
+
+        public SmsLinkResponse SendSmsLink(SmsLinkOtpRequest request, bool errorOnAccountStatus = false)
+        {
+            string[] validFactorIds = { "Phone1", "Phone2", "Phone3", "Phone4" };
+
+            // sanitize request
+            if (string.IsNullOrEmpty(request.UserId))
+            {
+                throw new ArgumentNullException("SendSmsLink.UserId", "User ID cannot be empty.");
+            }
+            if (string.IsNullOrEmpty(request.FactorId))
+            {
+                throw new ArgumentNullException("SendSmsLink.FactorId", "FactorId cannot be empty.");
+            }
+            if (!validFactorIds.Contains(request.FactorId))
+            {
+                throw new ArgumentException("Invalid FactorId.", "SendSmsLink.FactorId");
+            }
+
+            string apiVersion = errorOnAccountStatus ? ApiVersion.V1.Value : ApiVersion.V2.Value;
+
+            // process request
+            return this._apiClient.Post<SmsLinkResponse>("/api/" + apiVersion + "/auth", request);
         }
 
 
+        public EmailLinkResponse SendEmailLinkOtp(EmailLinkOtpRequest request, bool errorOnAccountStatus = false)

```

Comment:
SmsLink and EmailLink can also be sent via Adhoc. Can you add adhoc method for these options too please